src/main/scala/mesosphere/marathon/core/task/Task.scala

```diff
@@ -122,7 +122,7 @@ object Task {
     private val LegacyTaskIdRegex = """^(.+)[\._]([^_\.]+)$""".r
 
     // Regular expression for matching taskIds since instance-era
-    private val TaskIdWithInstanceIdRegex = """^(.+)\.(instance-|marathon-)([^_\.]+)[\._]([^_\.]+)$""".r
+    private val TaskIdWithInstanceIdRegex = """^(.+)\.(instance-|marathon-)([^_\.]+)[\._](.+)$""".r
```

Comment:
Should we add a comment here that explains why the container name group allows everything and that we expect validation taking place in other places?